prov/udp/src/udpx_ep.c

```diff
@@ -56,11 +56,25 @@ int udpx_setname(fid_t fid, void *addr, size_t addrlen)
 int udpx_getname(fid_t fid, void *addr, size_t *addrlen)
 {
 	struct udpx_ep *ep;
+	struct sockaddr_storage sock_addr = { 0 };
+	size_t sock_addr_len = sizeof(sock_addr);
 	int ret;
 
 	ep = container_of(fid, struct udpx_ep, util_ep.ep_fid.fid);
-	ret = getsockname(ep->sock, addr, (socklen_t *)addrlen);
-	return ret ? -errno : 0;
+	ret = getsockname(ep->sock, (struct sockaddr *)&sock_addr,
+			  (socklen_t *)&sock_addr_len);
+	if (ret)
+		return -ofi_sockerr();
+
+	if (*addrlen < sock_addr_len)
+		ret = -FI_ETOOSMALL;
+
+	*addrlen = sock_addr_len;
+
+	if (addr)
+		memcpy(addr, &sock_addr, *addrlen);
+
+	return ret;
```

Comment:
Please separate this change into its own patch, so that the fix can be cherry-picked into stable branches.

Reply:
Never mind -- I thought I was looking at a single commit, rather than the change set.

Reply:
:)